lib/ansible/modules/network/nxos/nxos_file_copy.py

```diff
@@ -218,28 +356,40 @@ def main():
     file_system = module.params['file_system']
 
     results['transfer_status'] = 'No Transfer'
-    results['local_file'] = local_file
     results['file_system'] = file_system
 
-    if not local_file_exists(module):
-        module.fail_json(msg="Local file {0} not found".format(local_file))
+    if file_pull:
+        src = remote_file.split('/')[-1]
+        local = local_file or src
 
-    dest = remote_file or os.path.basename(local_file)
-    remote_exists = remote_file_exists(module, dest, file_system=file_system)
+        if not module.check_mode:
+            copy_file_from_remote(module, local, file_system=file_system)
+            results['transfer_status'] = 'Received'
 
-    if not remote_exists:
         results['changed'] = True
-        file_exists = False
+        results['remote_file'] = src
+        results['local_file'] = local
     else:
-        file_exists = True
+        if not local_file_exists(module):
+            module.fail_json(msg="Local file {0} not found".format(local_file))
+
+        dest = remote_file or os.path.basename(local_file)
+        remote_exists = remote_file_exists(module, dest, file_system=file_system)
+
+        if not remote_exists:
+            results['changed'] = True
+            file_exists = False
+        else:
+            file_exists = True
 
-    if not module.check_mode and not file_exists:
-        transfer_file(module, dest)
-        results['transfer_status'] = 'Sent'
+        if not module.check_mode and not file_exists:
+            transfer_file_to_device(module, dest)
```

Comment:
Can you please change the term `device` to `controller` here? Also in the doc.
Using device can be confusing here as it might mean network device.